### ISSUE TITLE
Optionally use `hiera_hash` on `telegraf::inputs` and `telegraf::outputs`

### ISSUE THREAD

**Issue by joshuaspence** [voxpupuli/puppet-telegraf#27]
It would be convenient if this module could do something similar to https://forge.puppet.com/yo61/logrotate, which exposes a `$hieramerge` parameter:

``` puppet
# apply user-defined rules
class logrotate::rules{

  $hieramerge = $::logrotate::hieramerge
  $rules      = $::logrotate::rules

  if $hieramerge {
    $_rules = hiera_hash('logrotate::rules', $rules)
  } else {
    $_rules = $rules
  }

  create_resources('logrotate::rule', $_rules)

}
```

Currently, we are using the following workaround:

``` puppet
  class { 'telegraf':
    inputs  => hiera_hash('telegraf::inputs'),
    outputs => hiera_hash('telegraf::outputs'),
  }
```

This workaround is required because Puppet 3's automatic parameter lookup doesn't support deep merging.

**Comment by yankcrime** [comment]
Fixed in #25.

**Comment by joshuaspence** [comment]
I dont think that #25 actually fixes this... If I set a value for `telegraf::inputs` (or `telegraf::outputs`) then the value from `telegraf::params` isn't read.
